packages/lwc-language-server/src/javascript/type-mapping.ts

```diff
@@ -0,0 +1,560 @@
+                return [key, classProperty];
+            })
+        );
+    }
+    return {
+        staticObj,
+        params,
+    };
+}
+
+function getDecoratedWiredMethod(method: ClassMethod, decorator: WireDecorator): WireDecoratorTarget {
+    const { staticObj, params } = getWireParams(decorator);
+
+    const adapter = {
+        name: decorator.adapterId.localName,
+        reference: decorator.adapterModule,
+    };
+
+    return {
+        type: 'method',
+        name: method.name,
+        static: staticObj,
+        params,
+        // The old LWC metadata included an 'adapter' property, which
+        // is not used by lightning-language-server and is not captured in
+        // the WireDecoratorTarget type. But we've captured it here so that
+        // the output exactly matches that of the old LWC compiler's metadata.
+        adapter,
+    };
+}
+
+function getDecoratedMethods(methods: ClassMethod[]): {
+    wiredMethods: WireDecoratorTarget[],
+    apiMethods:  ApiDecoratorTarget[],
+    methodLocs: Map<string, number>,
+} {
+    const wiredMethods: WireDecoratorTarget[] = [];
+    const apiMethods: ApiDecoratorTarget[] = [];
+    const methodLocs: Map<string, number> = new Map();
+
+    for (const method of methods) {
+        const decorator = getDecorator(method.decorators, method);
+        if (!decorator) {
+            continue;
+        }
+
+        // This information is later used to reorder the combination of
+        // methods and properties, so that the order matches their original
+        // locations in the component code.
+        methodLocs.set(method.name, method.location.start);
+
+        if (decorator.type === 'Api') {
+            apiMethods.push(getDecoratedApiMethod(method));
+        } else if (decorator.type === 'Wire') {
+            wiredMethods.push(getDecoratedWiredMethod(method, decorator));
+        }
+    }
+
+    return {
+        wiredMethods,
+        apiMethods,
+        methodLocs,
+    };
+}
+
+function getDecoratedApiProperty(prop: ClassProperty): ApiDecoratorTarget {
+    return stripKeysWithUndefinedVals({
+        name: prop.name,
+        type: 'property',
+        value: dataPropertyToPropValue('api', prop.dataProperty),
+    });
+}
+
+function getDecoratedWiredProperty(prop: ClassProperty, decorator: WireDecorator): WireDecoratorTarget {
+    const { staticObj, params } = getWireParams(decorator);
+
+    const adapter = {
+        name: decorator.adapterId.localName,
+        reference: decorator.adapterModule,
+    };
+
+    return {
+        name: prop.name,
+        type: 'property',
+        static: staticObj,
+        params,
+        // The old LWC metadata included an 'adapter' property, which
+        // is not used by lightning-language-server and is not captured in
+        // the WireDecoratorTarget type. But we've captured it here so that
+        // the output exactly matches that of the old LWC compiler's metadata.
```

Comment:
To add to this comment, getting the `adapter` value is extremely inexpensive (as seen in the lines above). Because of that, it seemed worthwhile to include here with a corresponding change in the `WireDecoratorTarget` type. This allows for exact duplication of old metadata, making it easier to assert equivalency.